### ISSUE TITLE
Version 1.0 - does not work version 1.1.30 

### ISSUE THREAD

**Issue by VertexML** [Cacti/plugin_reportit#9]
Plugin loads fine - Cacti 1.1.30 
Cant create first or any report template, or report
Error: Unable to unlock this template without defined Measurands.
No error in load when debug plugin

Paradox: No Measurand edit link (until the first report created - typically first report is locked by default V0.75a to support this)
I tried using the direct URL to add a Measured, link but it errors also to create a measurand manually.

No knowledge prior versions of Cacti V1+ - but before reportIt worked great!
First time I have tried use V1.x Cacti

V0.75a worked create for many years, but now looking to migrate all to Cacti 1+
I know I wont be able to import old reports, but willing to rebuild them on new V1.0

Please confirm which version cacti reportit 1.0 was last tested. 

Is known and previously tested to work Cacti V1.x to work ??
Maybe I can use lower version of Cacti perhaps?

Thanks,

**Comment by SergejLT** [comment]
Was great plugin on older cacti version v0.8... Helped alot tracking statistics of hundreds DS day by day.
They should have implemented into cacti v1 by default.
sad that this project is abandoned...

**Comment by VvCD9ytqJaYBnbB5T6bdA** [comment]
This problem was also encountered in the recent test, which was not available at all in the latest version of cacti. I can't use it even if I add it directly to the database.
Maybe cacti has given up the project.
Hope can be solved.

**Comment by browniebraun** [comment]
I'm really sorry. I know this project looks more like a new version of Duke Nukem forever.
Please see my latest update. The plugin itself is not functional so far, but I would be really happy if someone could spend a few minutes reviewing the new report template part.
-Andi

**Comment by VertexML** [comment]
Hey, Duke Nukem or Doom comeback I would play. : )

Wow! thanks so much. I checked and update/new report template and seams mostly OK. I never saw groups before, and also never saw being able to combine two different data sources in one template before. I have only used up to 0.7.5a.   

I tried checking various things for you if it helps:

I get an error immediately adding a second group. First group was Cisco cpu added to template (one DS), second group I add stock Interface traffic (two DSs - in/out)
```
Validation error for variable with a value of . See backtrace below for more details.
--
(/plugins/reportit/templates.php: 66 template_edit)(/plugins/reportit/templates.php: 1105 draw_tab_table)(/plugins/reportit/templates.php: 1544 die_html_input_error)(/lib/html_validate.php: 60 cacti_debug_backtrace)

```
But after errors I can still see second group. Then can edit from Generic name duplicate of first. S ignoring error it appears it still works.

Variables tab seams to work include some error validation, duplicate function, delete etc.

Measurands: the only thing I can find is that 'Formatting' section:

The precision or Prefixed drop lists never appears - on any drop list selection.
In 0.7.5a  it was Base1000, base1024, or OFF (for integers and floating point items items)
With exception Scientific which greyed  out precision drop list

Ahh!  I just checked IE11. Different behavior. Both precision or Prefixed drop lists is about 3 characters wide. Seams to drop down and grey/ungrey out - but cant read text. Chrome was thin line, like was not even there. checking again, it does drop down a full box on clicking the thin line. Google Version 63.0.3239.132 (Official Build) (64-bit) - thin line. Funny, same thin link on IE Edge

Double Ahh! - I am using theme classic. I changed to Modern which both drop lists works fine. 

Which is the Theme people mostly test Cacti 1.0??  I can see themes in 1.0 are going to be a bit of pain for testers..

Cheers,

Jay

**Comment by browniebraun** [comment]
Many thanks for your great feedback. :)
Validation warning should have gone now. I hope, once this plugin is ready, people will enjoy the advantage of having data template groups. 
Regards
-Andi

**Comment by SergejLT** [comment]
my setup: cacti v1.1.30, modern theme, firefox.

Creating report templates seems to me working jus fine. Even when creating more than one group or measurands. 
I noticed that in "Data Templates" some assigned data template only selects 1 DS Item (cant add more that one DS Item when clicking on assigned group). Others are good like "Interface - Traffic" selects 2/2, "Interface - Errors" selects 4/4 (I can deselect DS items)

for example

![cacti1](https://user-images.githubusercontent.com/32635039/34973782-2a290b58-fa92-11e7-8c9c-8bca516e7a71.PNG)

"Local - Poller statistics" has DS items ->

![cacti2](https://user-images.githubusercontent.com/32635039/34973786-31509e64-fa92-11e7-99c5-bdf7bcfe5d4c.PNG)


Keep the good job!

**Comment by browniebraun** [comment]
On my lab device I currently have the set of default templates being installed. I've just added all data templates to a report template, but everything works as it should. Could you probably export and share some of the templates being affected?
Did you notice any error messages?
Regards
-Andi

**Comment by SergejLT** [comment]
No errors with debug mode.

[cacti_data_template_local_-_poller_statistics.txt](https://github.com/Cacti/plugin_reportit/files/1634210/cacti_data_template_local_-_poller_statistics.txt)

**Comment by browniebraun** [comment]
Table definition for column ID was not correct (too small).
ALTER TABLE `plugin_reportit_data_source_items` CHANGE `id` `id` mediumint(8) unsigned not null default '0';

**Comment by SergejLT** [comment]
nice! chagned and now its working.
tested if other custom data templates. everything seems to be ok

**Comment by SergejLT** [comment]
What's the status with the plugin?

**Comment by cigamit** [comment]
Rony is going to attempt to spend some time merging a legacy branch into develop.  The legacy branch seems to be pretty much functional.  We will move the current develop branch into a feature branch to take the pressure off of browniebraun.

**Comment by VertexML** [comment]
Was away on leave. Come back and find this now working!  Well done guys, you are great!!
Seams all working for my needs. Closing this now, but hope to provide user test feedback as I go.
